rust/pact_verifier/src/messages.rs

```diff
@@ -42,11 +43,12 @@ pub async fn verify_message_from_provider<F: RequestFilterExecutor>(
   };
   match make_provider_request(provider, &message_request, options, client).await {
     Ok(ref actual_response) => {
+      log::debug!("actual response = {:?}", actual_response);
+
+      let metadata = extract_metadata(actual_response);
       let actual = Message {
         contents: actual_response.body.clone(),
```

Comment:
See comment directly on the PR - this doesn't match the draft implementation on the client (which does match your proposed improvement ;) ).

Reply:
yep. I think I'm going to make it accept an object that wraps both the message and the metadata, or fallback to current behaviour. That way we don't break existing clients.